client/python/lib/test/test_tensors.py

```diff
@@ -237,96 +353,135 @@ def test_make_tensor_proto_valid_make_ndarray_valid():
 
-def test_make_tensor_proto_valid_no_dtype_provided():
-    values = [1.0, 2.0, 3.0]
-    tensor_proto = make_tensor_proto(shape=[3], values = values)
-    np_values = np.array(values)
-    assert tensor_proto.tensor_content == np_values.tobytes()
-    assert tensor_proto.dtype == DataType.DT_DOUBLE
-    assert tensor_proto.tensor_shape == TensorShapeProto(dim = [TensorShapeProto.Dim(size=3)])
+def test_make_tensor_proto_valid_string_reshape():
+    values = bytes([0x13, 0x00, 0x00, 0x00, 0x08, 0x00])
+    tensor_proto = make_tensor_proto(values = [[values, values],[values, values]], shape=[4], dtype = DataType.DT_STRING)
+    assert all(val == np.array(values) for val in tensor_proto.string_val)
+    assert tensor_proto.dtype == DataType.DT_STRING
+    assert tensor_proto.tensor_shape == TensorShapeProto(dim = [TensorShapeProto.Dim(size=4)])
 
-def test_make_tensor_proto_valid_no_shape_provided():
-    values = [[1.0, 2.0, 3.0], [1.0, 2.0, 3.0]]
-    tensor_proto = make_tensor_proto(dtype = DataType.DT_FLOAT, values = values)
-    expected_shape = TensorShapeProto(dim = [TensorShapeProto.Dim(size=2), TensorShapeProto.Dim(size=3)])
-    np_values = np.array(values, np.float32)
-    assert tensor_proto.tensor_content == np_values.tobytes()
-    assert tensor_proto.tensor_shape == expected_shape
-    assert tensor_proto.dtype == DataType.DT_FLOAT
+def test_make_tensor_proto_valid_2D_array_to_bytes():
+    values = [[1,2,3],[4,5,6]]
+    tensor_proto = make_tensor_proto(values = values, shape=[6], dtype = DataType.DT_STRING)
+    assert tensor_proto.string_val == np.array(values, dtype=np.bytes_).reshape([6]).tolist()
+    assert tensor_proto.dtype == DataType.DT_STRING
+    assert tensor_proto.tensor_shape == TensorShapeProto(dim = [TensorShapeProto.Dim(size=6)])
 
-# def test_make_tensor_proto_invalid_dtype_provided():
-#     values = [[1.0, 2.0, 3.0], [1.0, 2.0, 3.0]]
-#     with pytest.raises(TypeError) as exception_info:
-#         make_tensor_proto(shape = None, dtype = DataType.DT_STRING, values = values)
-#         exception = exception_info.value
-#         assert str(exception) == "[1.0, 2.0, 3.0] has type list, but expected one of: bytes"
+def test_make_tensor_proto_invalid_shape_element_values():
+    values = [1,2,3]
+    with pytest.raises(ValueError) as exception_info:
+        make_tensor_proto(values = values, shape=[-3], dtype = None)
+    exception = exception_info.value
+    assert str(exception) == "shape elements should be unsigned integers"
 
-def test_make_tensor_proto_invalid_dimsions():
+def test_make_tensor_proto_invalid_dimensions():
     values = [[1.0, 2.0], [1.0, 2.0, 3.0]]
     with pytest.raises(ValueError) as exception_info:
-        make_tensor_proto(shape=[2,3], dtype = DataType.DT_FLOAT, values = values)
-        exception = exception_info.value
-        assert str(exception) == "setting an array element with a sequence"
+        make_tensor_proto(values = values, shape=[2,3], dtype = DataType.DT_FLOAT)
```

Comment:
The issue here is with asarray()? not np.array() call on array creation?
The error message does not make it clear that there's a problem with dimensions.